src/ucp/wireup/wireup_cm.c

```diff
@@ -849,7 +851,7 @@ ucs_status_t ucp_ep_client_cm_connect_start(ucp_ep_h ucp_ep,
     ucs_status_t status;
 
     wireup_ep->ep_init_flags = ucp_ep_init_flags(ucp_ep->worker, params);
-    wireup_ep->cm_idx        = 0;
+    wireup_ep->cm_idx        = ucp_ep_config(ucp_ep)->key.cm_idx;
```

Comment:
maybe we can remove cm_idx field from wireup_ep altogether?

Reply:
we should be able to take cm_idx from current ep config key: when ucp_worker_create_ep() is called from ucp_ep_create_to_sock_addr it should set the cm_idx to 0

Reply:
Yes, but if we do fallback, `ucp_cm_client_priv_pack_cb()` could be called several times for each CM until CM connection is successful.
So, we don't have the way to determine whether it was called after initial config get (CM idx has to be 0) or from fallback (CM idx has to be ++).
Whatd do you think about not initializing CM index to 0 when initializing config for the first time? Initialize it or increment it only in `ucp_cm_client_priv_pack_cb()`?

Reply:
done in another way:
saved CM index in EP extension